nixos/modules/services/hardware/bluetooth.nix

```diff
@@ -28,18 +31,25 @@ in
     environment.systemPackages = [ bluez-bluetooth pkgs.openobex pkgs.obexftp ];
 
     services.udev.packages = [ bluez-bluetooth ];
-
     services.dbus.packages = [ bluez-bluetooth ];
+    systemd.packages       = [ bluez-bluetooth ];
+
+    services.udev.extraRules = optionalString config.hardware.bluetooth.powerOnBoot ''
+      ACTION=="add", KERNEL=="hci0", RUN+="${pkgs.bluez.out}/bin/hciconfig %k up"
+    '';
 
-    systemd.packages = [ bluez-bluetooth ];
+    systemd.services = {
+      bluetooth = {
+        wantedBy = [ "bluetooth.target" ];
+        aliases  = [ "dbus-org.bluez.service" ];
+      };
 
-    systemd.services.bluetooth = {
-      wantedBy = [ "bluetooth.target" ];
-      aliases = [ "dbus-org.bluez.service" ];
+      systemd-udevd.serviceConfig.RestrictAddressFamilies =
+        "AF_UNIX AF_NETLINK AF_INET AF_INET6 AF_BLUETOOTH";
```

Comment:
I would prefer a service with the correct permissions rather then changing security defaults of udevd. A service can be started from udev using:

```
 ACTION=="add", KERNEL=="hci0", ENV{SYSTEMD_WANTS}="power-up-bluetooth.service"
```